### ISSUE TITLE
The input line does not match the specified log format.Line

### ISSUE THREAD

**Issue by lingxiao26** [nielsbasjes/logparser#60]
Hello, sorry to bother you~. I am not very good at English. I wish you can understand what I say o(╯□╰)o .

I use your logparser to parse k8s ingress log. This is my log pattern

**PATTERN**
```
"$the_real_ip - [$the_real_ip] - $remote_user [$time_local] \"$request\" $status $body_bytes_sent \"$http_referer\" \"$http_user_agent\" $request_length $request_time [$proxy_upstream_name] $upstream_addr $upstream_response_length $upstream_response_time $upstream_status $req_id"
```

when I use this pattern to parse. I may got this Exception "nl.basjes.parse.core.exceptions.DissectionFailure".
Maybe because some of the log line not Exact match the pattern

**Exception Message**
```
Exception in thread "main" nl.basjes.parse.core.exceptions.DissectionFailure: The input line does not match the specified log format.Line     : Value{filled=STRING, s='112.64.166.2 - [112.64.166.2] - - [07/Aug/2020:15:50:07 +0800] "HEAD /ai/search/version HTTP/1.1" 308 0 "-" "curl/7.29.0" 100 0.000 [default-ai-search-prod-svc-5009] - - - - bfb9417db656d95bcfdf3e2a7f47b1ec', l=null, d=null}
```

I just want `request field` like `HEAD /ai/search/version HTTP/1.1`.
So, is there a way to avoid this Exception

Finally, thanks supply this lib.
Best wishes!

**Comment by nielsbasjes** [comment]
Thanks for your report.
I was able to reproduce the problem with your input.
I am working on a solution.

**Comment by lingxiao26** [comment]
This is my first issue in github. I feel a little bit amazing , haha!!
Thanks your guys for supply so many excellent open source project. 
It's really helpful to me.

**Comment by nielsbasjes** [comment]
I think I fixed it.
Can you please test?
I made a test build that you can find in the maven repository   https://oss.sonatype.org/content/repositories/snapshots/
The version is 5.4-SNAPSHOT

I am looking forward for your feedback if this is correct now.
If you have more / different examples of problems I would really like to have them.

**Comment by lingxiao26** [comment]
Thank you very mush.
I use your library in the company project. 
I'm home now, and I can't connect to company network
I will test it tomorrw.
Thanks again!!!

**Comment by lingxiao26** [comment]
By the way, I think we have a different time .
Now, my time is `Sat Aug  8 01:00:19 CST 2020`

**Comment by nielsbasjes** [comment]
Yes, it is 19:09 here right now.

**Comment by lingxiao26** [comment]
Sorry, could you please supply a config in pom.xml like this
```
      <dependency>
            <groupId>nl.basjes.parse.httpdlog</groupId>
            <artifactId>httpdlog-parser</artifactId>
            <version>5.3</version>
        </dependency>
```
I don't know how to specify the repository that you supply.
My maven config `setting.xml ` specified a private nexus url.
If dependency not found, it will get the dependency from maven repository  `https://mvnrepository.com/`

**Comment by nielsbasjes** [comment]
In your pom.xml you should put something like this:
```
  <repositories>
    <repository>
      <id>sonatype-snapshots</id>
      <name>Sonatype Snapshots</name>
      <snapshots><enabled>true</enabled></snapshots>
      <releases><enabled>false</enabled></releases>
      <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
    </repository>
  </repositories>

  <dependencies>
    <dependency>
      <groupId>nl.basjes.parse.httpdlog</groupId>
      <artifactId>httpdlog-parser</artifactId>
      <version>5.4-SNAPSHOT</version>
    </dependency>
  </dependencies>
```

**Comment by lingxiao26** [comment]
It's still have some log didn't match pattern.
I put the error log to the attachment, you can download it.

error line: 96
total log line: 34419

[errorlog.txt](https://github.com/nielsbasjes/logparser/files/5044788/errorlog.txt)

**Comment by nielsbasjes** [comment]
These are not access log lines. 
These look like generic application logging.

**Comment by lingxiao26** [comment]
It's comes from k8s ingress pod `nginx-ingress-controller-55dffd9cf5-8swdg`

**Comment by nielsbasjes** [comment]
It looks like you are getting (at least) two different types of logging mixed together.
You must separate them and process them independently.
Or put a filter to split them in the processing (i.e. if (line.contains("[warn]")) )

**Comment by lingxiao26** [comment]
I catched the Exception, and just do nothing, ignore that not match log.
Sorry for wasted your time.
Respect !!

**Comment by nielsbasjes** [comment]
Ok, so the normal lines are now working correctly for you?

**Comment by lingxiao26** [comment]
Yes, most of lines are working correctly.
Should I close this issue?

**Comment by nielsbasjes** [comment]
Yes, if it works for you.
I do think that in your current solution (catching the Exception) you can no longer see the difference between a "not access log line" and a "access log line which has a problem".

**Comment by lingxiao26** [comment]
OK!
Best wishes for you!

**Comment by nielsbasjes** [comment]
I released version 5.4 just now.
It usually takes an hour before you can use it.
